### PR TITLE
Add note that DSN can be public

### PR DESCRIPTION
**Pull request by mlissner**
One of the first questions I had while integrating Sentry is whether the DSN needed to kept private. I'm not sure the wording here is perfect, but this should help clarify for future people.

Details here: https://forum.sentry.io/t/dsn-private-public/6297/3

### CODE REVIEW

**Comment by mlissner** [comment]
Anybody interested in reviewing this tiny documentation PR?

**Comment by PeloWriter** [comment]
I'd suggest we publish this here - /product/sentry-basics/dsn-explainer/ to help folks who need more detail regarding our DSN.

Thank you.

**Comment by mlissner** [comment]
Oh no! On Tuesday we had a huge scare at my organization due to the DSN being public, and I fully regret this PR. I wrote a long note to your support folks yesterday explaining the whole thing. This PR is exactly backwards and I think we should change it accordingly. Here's what I wrote to your support folks:

> All of our code is open source. About six months ago, when we set up Sentry, we debated if we should keep the DSN private or just include it in our source code on Github. After checking the docs and thinking it through, we concluded that the biggest risk was that somebody could send an error to our Sentry project via a dev machine or via the API and we'd have to sort that out. Seemed low risk and unlikely to happen, so we went ahead and included our DSN in our code and released it. We put it behind a setting so that in theory, dev machines wouldn't send errors.

> Alas, yesterday, it happened! Somebody cloned our code and tweaked their settings so that their machine was configured as a production server instead of a developer machine. That tweak enabled Sentry, and then that person generated [an error]. The error indicated that somebody had run a program on the command line that would normally create a Django superuser. The program crashed for some reason, and we got a Sentry issue.

> Making matters worse, because we use Docker, although we noticed that the server associated with the issue wasn't one we were familiar with (it was just a random number), we couldn't determine whether the issue had occurred on a docker image in prod that had since been deleted. Meaning we couldn't be sure it happened on a dev machine or not.

> We spent much of the day fretting yesterday, trying to learn if we had an intruder running shell commands, and attempting to respond. It was a horrible day. Eventually, I realized that I could look at other issues from that same server in Sentry, and I found a few others that had very clear indicia that they had come from a developer machine, so that was that. One terrible day lost, but no intruder, thank goodness. Last night we contacted a new developer that has been filing issues on Github to ask if they did this, and they confirmed they did (we are in CA and they are in Germany, so this took some time).

> I think this has a few lessons for both sides. On our side, we've revoked the DSN that was in the public source code, generated a dev project in Sentry, and put that DSN in our source code instead. In prod, we now have a setting for the DSN that we keep private. Great.

And the important part from my message:

> On your side, I think this is a documentation issue, or perhaps I've missed it, but I don't think there's much documentation saying that the DSN should be kept private. Based on this, before yesterday, we thought it was fine to have it be public, but, though that can't cause any real problems, it sure can be disruptive! I'd suggest that your docs and various installation instructions should make this more clear.

**Comment by PeloWriter** [comment]
Thanks for writing back. I'm so glad you shared this so I can fully understand the context. I'll review further with the team.

File: src/collections/_documentation/error-reporting/configuration/index.md [hunk 1]
```diff
@@ -24,6 +24,9 @@ events.
 
 In runtimes without a process environment (such as the browser) that fallback does not apply.
 
+The DSN does not need to be kept private. If it is abused, the worst case is that another 
+application could send events to your account.
```
[the remark this answers is not in the document]
Comment (mlissner) [reply src/collections/_documentation/error-reporting/configuration/index.md:28]: I don't like how that forum post is framing this as a binary thing. There are shades of gray between fully public and fully private. After my experience, I'd strongly suggest that the policy should be something like, "The DSN should be kept private *to the extent possible*."

I agree that the worst case is that somebody can send a message to Sentry, but I don't think folks realize just how bad that can be. In our case, it was terrifying and sent our organization into a tailspin until we could figure out what was going on. 

I'd also note that it's possible to have different levels of privacy for different DSNs. For example, the ones that are in your JavaScript can be expected to have the occasional false message from an abuser, and your back end can be set up with a different DSN that's kept totally private. 

This kind of documentation would have helped us a lot and isn't that hard to understand.

**Comment by dcramer** [comment]
I think we should generally call out DSN utilization. Here is some general direction:

1) DSNs are safe to keep public - they only allow submission of new events and related event data. They do not allow read access to any information.

2) There is a risk of abuse on DSNs, where any user can send events to your organization with any information they want. This cannot be avoided, but we provide controls to block IPs and similar concerns if the situation happens. You can also rotate (and revoke) DSNs in these kinds of scenarios.

3) For applications which are shipped to client devices, if possible, we recommend having a way to configure the DSN dynamically. In an ideal scenario you can "ship" a new DSN to your application without the customer downloading the latest version. This may not always be practical and we cannot offer advice as its implementation specific.

I'll note that #2 rarely happens in practice. I do agree w/ the cloning scenario being a non-obvious risk to some people, so we can call that out as well.

**Comment by PeloWriter** [comment]
Super good discussion here, thanks for raising this point and persevering to clarity @mlissner. Since this branch is terrible stale, I've opened a [fresh PR](https://github.com/getsentry/sentry-docs/pull/3123) to:
- add utilization to our explanation of DSN
- added a pointer to that information from our configuration content

**Pull request closed without merging**